src/scripts/clipperUI/panels/signInPanel.tsx

```diff
@@ -60,15 +84,43 @@ class SignInPanelClass extends ComponentBase<{}, SignInPanelProps> {
 							{signInDescription}
 						</span>
 					</div>
-					<a id={Constants.Ids.signInButtonMsa} {...this.enableInvoke(this.onSignInMsa, 1)}>
+					{signInFailureDetected
```

Comment:
Consider moving each of these chunks into a method for readability.
simple example

```
<div>
   {this.getElemOne()}
   {this.getElemTwo()}
   {this.getElemThree()}
</div>
```